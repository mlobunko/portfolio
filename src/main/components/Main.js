import React from "react";

import Greetings from "app/greetings/components";
import ProjectPhoto from "app/projectPhoto";
import Skills from "app/skills/components";
import Footer from "app/footer/components";

import ErrorBoundary from "./ErrorBoundary";
import { StyledHeaderOfSections, StyledMain } from "./styles";
import ImageWithoutWater from "../images/image-without-water.jpg";
import ImageFunnyFarm from "../images/image-funny-farm.jpg";
import ImageMovies from "../images/image-movies.jpg";
import ImagePomodoro from "../images/image-pomidoro.jpg";
import {
  descriptionPomodoroTimer,
  descriptionWithoutWater,
  descriptionFunnyFarm,
  descriptionMovies
} from "../constants";

export const Main = () => (
  <StyledMain>
    <ErrorBoundary>
      <Greetings />
      <StyledHeaderOfSections>My Recent Projects</StyledHeaderOfSections>
      <ProjectPhoto
        description={descriptionPomodoroTimer}
        image={ImagePomodoro}
        textColor={"white"}
        link={"https://github.com/mlobunko/pomodoro-timer"}
        nameOfProject={"Pomodoro Timer"}
        margin={true}
        textUp={true}
      />
      <ProjectPhoto
        description={descriptionWithoutWater}
        image={ImageWithoutWater}
        textColor={"white"}
        link={"https://github.com/mlobunko/without-water"}
        nameOfProject={"Without Water"}
        textUp={true}
        margin={true}
      />
      <ProjectPhoto
        description={descriptionMovies}
        image={ImageMovies}
        textColor={"white"}
        link={"https://github.com/mlobunko/movies"}
        nameOfProject={"Movies"}
        margin={true}
      />
      <ProjectPhoto
        description={descriptionFunnyFarm}
        image={ImageFunnyFarm}
        link={"https://github.com/mlobunko/funny-farm"}
        nameOfProject={"Funny Farm"}
        textColor={"black"}
        textUp={true}
      />

      <StyledHeaderOfSections>My Skills</StyledHeaderOfSections>
      <Skills />
      <Footer />
    </ErrorBoundary>
  </StyledMain>
);

Main.displayName = "Main";

export default Main;
