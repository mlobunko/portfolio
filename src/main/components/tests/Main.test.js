import React from "react";
import { shallow } from "enzyme";

import { Main } from "../Main";

describe("render", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
