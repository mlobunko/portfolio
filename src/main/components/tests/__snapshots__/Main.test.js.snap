// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledMain>
  <ErrorBoundary>
    <Greetings />
    <StyledHeaderOfSections>
      My Recent Projects
    </StyledHeaderOfSections>
    <ProjectPhoto
      description="An app uses the Pomodoro Technique which breaks down work into intervals, traditionally 25 minutes in length, separated by short breaks."
      image="image-pomidoro.jpg"
      link="https://github.com/mlobunko/pomodoro-timer"
      margin={true}
      nameOfProject="Pomodoro Timer"
      textColor="white"
      textUp={true}
    />
    <ProjectPhoto
      description="Project based on the list of tasks usually solve by the front-end developer."
      image="image-without-water.jpg"
      link="https://github.com/mlobunko/without-water"
      margin={true}
      nameOfProject="Without Water"
      textColor="white"
      textUp={true}
    />
    <ProjectPhoto
      description="An application for search movie details uses The Movie Database."
      image="image-movies.jpg"
      link="https://github.com/mlobunko/movies"
      margin={true}
      nameOfProject="Movies"
      textColor="white"
    />
    <ProjectPhoto
      description="A sound application plays voices of animals and shows their kind."
      image="image-funny-farm.jpg"
      link="https://github.com/mlobunko/funny-farm"
      nameOfProject="Funny Farm"
      textColor="black"
      textUp={true}
    />
    <StyledHeaderOfSections>
      My Skills
    </StyledHeaderOfSections>
    <Skills />
    <Component />
  </ErrorBoundary>
</StyledMain>
`;
